### PR TITLE
feat: Drawer accept translateY param, timeout will reset when timerLe…

### PR DESCRIPTION
**Pull request by Jacky-Rolo**
## feature
`Drawer` Component
- accept `translateY` parameter. `transform` style depend on `translateY` prop instead `height`
- `useEffect setTimeout` will trigger when `timerLength` change
  - let's able to reset `timeout` from outside 
- accept `panResponderHandlers` parameter, let Drawer are able to use gesture

### basic `Drawer` (without timeout)
- `Drawer` open/close only when user click button

### Timeout `Drawer` (with timeout)
- `Drawer` open/close when timeout
- `Drawer` will keep **open** when user has interaction on `Drawer`

### `PanResponder Drawer`
- click `toggle button` to test control `animatedTranslateY`
- drag **drag here** to test drag.

## screenshots 
image  
image  
image  
image

### CODE REVIEW

**Review by FrontendPatriciaAtRolo — CHANGES_REQUESTED** [review]

File: src/components/Drawer/Drawer.js [hunk 1]
```diff
@@ -60,24 +68,36 @@ export const Drawer = (props) => {
...
       style={StyleSheet.flatten([classes.root, rootStyles])}
       onPress={() => setIsRecount(true)}
       {...rest}
     >
+      {panResponderHandlers && (
+        <Animated.View
+          style={StyleSheet.flatten([classes.panResponder, panResponderStyle])}
+          {...panResponderHandlers}
+        >
+          {RanResponderBar ? RanResponderBar : <View style={classes.panResponderBar} />}
```
[the remark this answers is not in the document]
Comment (Jacky-Rolo) [reply src/components/Drawer/Drawer.js:98]: I want to let user to use `RanResponderBar`. Pass **Component** to replace `<View style={classes.panResponderBar} />`   

so user can wrap bullets into `RanResponderBar`

File: src/components/Drawer/Drawer.styles.js [hunk 2]
```diff
@@ -21,8 +21,20 @@ export const useStyles = makeStyles(() => ({
       position: 'absolute',
       width,
       zIndex: 2,
-      transform: [{translateY: sliderAnimation.current}],
+      transform: [{translateY: sliderAnimation}],
       ...positionMixinStyle,
     };
   },
+  panResponder: {
+    width: '100%',
```
[the remark this answers is not in the document]
Comment (Jacky-Rolo) [reply src/components/Drawer/Drawer.styles.js:29]: I will double check this!! 
thanks

File: src/components/Drawer/Drawer.js [hunk 3]
```diff
@@ -15,6 +15,10 @@ import {useStyles} from './Drawer.styles';
  * @param {number} props.show - Display container or not
  * @param {func} props.onHide - Callback for container onHide
  * @param {node} props.children - Content to be included
+ * @param {func} props.panResponderHandlers - panResponder's panHandlers. Able to pass panHandlers to customize panResponder behavior.
+ * @param {func} props.animatedTranslateY - TranslateY (animated.value). Able to control "transform translateY" from outside.
+ * @param {func} props.panResponderStyle - PanResponder Element's style. Able to override PanResponder Element default style.
+ * @param {func} props.RanResponderBar - PanResponder Element's child. Able to customeize PanResponder's child.
```
Comment (FrontendPatriciaAtRolo) [inline src/components/Drawer/Drawer.js:21]: I am a bit confused now, coz you defined those props as an object, but also said it's a function.
image

**Review by FrontendPatriciaAtRolo — APPROVED** [review]
LGTM :)

File: src/components/Drawer/Drawer.js (hunk 3 above)
Comment (Jacky-Rolo) [reply src/components/Drawer/Drawer.js:21]: **GOOD CATCH**, my mistake. thanks

Comment (FrontendPatriciaAtRolo) [reply src/components/Drawer/Drawer.js]: Thank you, it's good to go :)

**Pull request merged**